### PR TITLE
11-topicQuery fully tested

### PR DESCRIPTION
**Pull request by euandecoup**

### CODE REVIEW

File: controllers/articles.controllers.js [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
         })
 }
 
 function getArticles(request, response, next) {
-    fetchAllArticles()
-        .then((articles) => {
+    const { topic } = request.query
+    promises = [fetchAllArticles(topic)]
+    if (topic) {
+        promises.push(checkTopicExists(topic))
```
Comment (nc-hannah-burks) [inline controllers/articles.controllers.js:19]: We don't want to risk SQL injection by potentially running fetchAllArticles(topic) before/same time as checking the topic exists. I would therefore make sure we do checkTopicExists and then we can do fetchAllArticles

Promise.all does not guarantee the order in which the promises are fulfilled.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -306,10 +297,35 @@ describe('App', ()=>{
...
+    describe('GET /api/articles (topic query)', () => {
+        test('GET 200: should return an array of article objects filtered by topic when topic query is provided', () => {
             return request(app)
-            .get('/api/userz')
-            .expect(404) 
+            .get('/api/articles?topic=cats')
+            .expect(200)
+            .then((response) => {
+                const articles = response.body.articles
+                expect(Array.isArray(articles)).toBe(true)
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:308]: I would also check length of array here :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -306,10 +297,35 @@ describe('App', ()=>{
...
+                })
+            })
+        });
+        test('GET 400: should respond with 400 for invalid topic query', () => {
+            return request(app)
+            .get('/api/articles?topic=123')
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('topic not found')
+            })
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:328]: It would be good to add another test for Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.

**Review by nc-hannah-burks — COMMENTED** [review]
A couple minor tweaks here Euan, but good work.
